.travis.yml

```diff
@@ -1,47 +1,37 @@
-language: python
+language: generic
+sudo: false
 
-sudo: true
+os:
+  - linux
+  - osx
 
-branches:
-  only:
-    - master
-
-addons:
-  apt:
-    sources:
-      - ubuntu-toolchain-r-test
-    packages:
-      - libstdc++-4.9-dev
-
-python:
-  # We don't actually use the Travis Python, but this keeps it organized.
-  - "2.7"
-  - "3.5"
-  - "3.6"
+env:
+  - PYTHON_VERSION=2.7 CONDA_VERSION=4.5
+  - PYTHON_VERSION=3.5 CONDA_VERSION=4.5
+  - PYTHON_VERSION=3.6 CONDA_VERSION=4.5
+  - PYTHON_VERSION=3.6 CONDA_VERSION=4.3
```

Comment:
What's the rationale for selecting this particular set of pythons/condas (especially the last one)? 
Also, if conda is pinned like this the test script will need to be updated every time there's a conda minor version update. Previously the repo was testing against the latest version in the default channel ( via `conda update conda --quiet --yes` ), but this is removed below (old line 34).

Reply:
@tjd2002, I'm trying to balance test coverage with not over-using Travis resources. I'm open to suggestion. But I do want at least one conda 4.3 test, because I know that we (Anaconda) need to confirm that this still works with that version of conda for awhile.

Reply:
Note that `conda install conda` and `conda update conda` are functionally equivalent. So we could do something like this:
```
PYTHON_VERSION=2.7 CONDA_SPEC=''
```
and then
```
conda install conda$CONDA_SPEC --yes --quiet
```
and this would "default" to the latest version of conda. What do you think?